src/mpi/coll/transports/mpich/api_def.h

```diff
@@ -0,0 +1,50 @@
+#define TRANSPORT_NAME MPICH_
```

Comment:
Why did we move all these into each transport? Having a common file makes it easier to add a new transport and it was pretty neat.